src/NHibernate/Linq/Visitors/NhExpressionVisitor.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Linq.Expressions;
+using NHibernate.Linq.Expressions;
+using Remotion.Linq.Parsing;
+
+namespace NHibernate.Linq.Visitors
+{
+	public class NhExpressionVisitor : RelinqExpressionVisitor
+	{
+		public override Expression Visit(Expression expression)
+		{
+			if (expression == null)
+			{
+				return null;
+			}
+
+			// Keep this variable for easy examination during debug.
+			var expr = base.Visit(expression);
+			return expr;
+		}
+
+		public virtual Expression VisitNhStar(NhStarExpression expression)
+			=> VisitExtension(expression);
```

Comment:
With [Expression Trees v2 Spec](http://www.codeplex.com/Download?ProjectName=dlr&DownloadId=246540), those methods are now called by `Expression.Accept`. The old default implementation (calling `Accept` on the expression) would cause an infinite calls loop. Instead, the default implementation should be now to call `VisitExtension`.